pylint/pyreverse/main.py

```diff
@@ -209,8 +207,7 @@ def run(self, args):
             diadefs = handler.get_diadefs(project, linker)
         finally:
             sys.path.pop(0)
-        printer_class = VCGPrinter if self.config.output_format == "vcg" else DotPrinter
-        writer.DiagramWriter(self.config, printer_class).write(diadefs)
+        writer.DiagramWriter(self.config).write(diadefs)
```

Comment:
I introduced a factory function to get the printer class for a given filetype. This supports the Open-Closed-Principle (only factory function needs to be updated if a new printer is added). Also, the ``DiagramWriter`` can now obtain the printer class itself.